sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/configuration/AadOAuth2ClientConfiguration.java

```diff
@@ -3,174 +3,160 @@
+import org.springframework.security.oauth2.client.endpoint.DefaultPasswordTokenResponseClient;
+import org.springframework.security.oauth2.client.endpoint.DefaultRefreshTokenTokenResponseClient;
+import org.springframework.security.oauth2.client.endpoint.OAuth2ClientCredentialsGrantRequestEntityConverter;
+import org.springframework.security.oauth2.client.endpoint.OAuth2PasswordGrantRequestEntityConverter;
+import org.springframework.security.oauth2.client.endpoint.OAuth2RefreshTokenGrantRequestEntityConverter;
 import org.springframework.security.oauth2.client.registration.ClientRegistration;
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
 import org.springframework.security.oauth2.client.web.DefaultOAuth2AuthorizedClientManager;
 import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
 
+import java.util.function.Function;
+
 /**
  * <p>
  * The configuration will not be activated if no {@link ClientRegistration} classes provided.
  * </p>
  */
 @Configuration(proxyBeanMethods = false)
+@Conditional(ClientRegistrationCondition.class)
 public class AadOAuth2ClientConfiguration {
 
     /**
-     * OAuth2 client configuration for AAD.
+     * Declare ClientRegistrationRepository bean.
+     *
+     * @param properties the AAD authentication properties
+     * @return ClientRegistrationRepository bean
      */
-    @Configuration(proxyBeanMethods = false)
-    @Conditional(ClientRegistrationCondition.class)
-    public static class OAuth2ClientRepositoryConfiguration {
-
-        /**
-         * Declare ClientRegistrationRepository bean.
-         *
-         * @param properties the AAD authentication properties
-         * @return ClientRegistrationRepository bean
-         */
-        @Bean
-        @ConditionalOnMissingBean
-        public ClientRegistrationRepository clientRegistrationRepository(AadAuthenticationProperties properties) {
-            return new AadClientRegistrationRepository(properties);
-        }
-
-        /**
-         * Declare OAuth2AuthorizedClientRepository bean.
-         *
-         * @return OAuth2AuthorizedClientRepository bean
-         */
-        @Bean
-        @ConditionalOnMissingBean
-        public OAuth2AuthorizedClientRepository oAuth2AuthorizedClientRepository() {
-            return new JacksonHttpSessionOAuth2AuthorizedClientRepository();
-        }
+    @Bean
+    @ConditionalOnMissingBean
+    public ClientRegistrationRepository clientRegistrationRepository(AadAuthenticationProperties properties) {
+        return new AadClientRegistrationRepository(properties);
     }
 
     /**
-     * Web application scenario, OAuth2AuthorizedClientManager configuration for AAD.
+     * Declare OAuth2AuthorizedClientRepository bean.
+     *
+     * @return OAuth2AuthorizedClientRepository bean
      */
-    @Configuration(proxyBeanMethods = false)
-    @Conditional(WebApplicationWithoutResourceServerCondition.class)
```

Comment:
same here

Reply:
Can we add more test cases for all conditions?

Reply:
Do we still need this `ResourceServerWithOboCondition`?

Reply:
Will remove the unused conditions